frontend/src/utils/tracing/UrlProviders/OpenShift.ts

```diff
@@ -0,0 +1,122 @@
+import { ConcreteService, TracingUrlProvider } from 'types/Tracing';
+import { ExternalServiceInfo, TempoConfig } from 'types/StatusState';
+import { BoundsInMilliseconds } from 'types/Common';
+import { SpanData, TraceData } from '../../../types/TracingInfo';
+
+interface OpenShiftExternalService extends ConcreteService {
+  name: string;
+  tempoConfig?: TempoConfig;
+}
+
+export function isOpenShiftService(svc: ExternalServiceInfo): svc is OpenShiftExternalService {
+  return svc.name.toLowerCase().includes('openshift') || svc.name.toLowerCase().includes('console');
+}
+
+export class OpenShiftUrlProvider implements TracingUrlProvider {
+  private service: OpenShiftExternalService;
+  readonly valid: boolean = true;
+
+  constructor(service: OpenShiftExternalService) {
+    this.service = service;
+  }
+
+  private buildBaseUrl(): string {
+    // Ensure the URL ends with a slash for proper path joining
+    const baseUrl = this.service.url.endsWith('/') ? this.service.url : `${this.service.url}/`;
+    return `${baseUrl}observe/traces`;
+  }
+
+  private buildQueryParams(bounds: BoundsInMilliseconds, tags: Record<string, string>, limit: number): string {
+    const params = new URLSearchParams();
+
+    // Add configurable parameters for OpenShift console
+    const namespace = this.service.tempoConfig?.namespace || 'tempo';
+    const name = this.service.tempoConfig?.name || 'sample';
+    const tenant = this.service.tempoConfig?.tenant || 'default';
+
+    params.append('namespace', namespace);
+    params.append('name', name);
+    params.append('tenant', tenant);
+    params.append('limit', limit.toString());
+
+    // Add time range parameters
+    if (bounds.from) {
+      params.append('start', (bounds.from * 1000).toString());
+    }
+    if (bounds.to && bounds.from) {
+      const duration = bounds.to - bounds.from;
+      // Convert duration to hours for the duration parameter
+      const durationHours = Math.max(1, Math.ceil(duration / 3600));
```

Comment:
Is this correct? If `bounds` is in millis then `duration` would be in millis. There are 3600 seconds in an hour but 3600000 milliseconds in an hour.